File: project_euler/Problem029/Problem029.java
package project_euler;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Problem029 {
    public void solution() {
        int start = 2;
        int end = 100;
        List<Double> terms = new ArrayList<>();
        for (int i = start; i <= end; ++i) {
            for (int j = start; j <= end; ++j) {
                terms.add(i * Math.log(j));
            }
        }

        Collections.sort(terms);

        int result = terms.size();
        for (int i = result - 1; i >= 1; --i) {
            if (Math.abs(terms.get(i) - terms.get(i - 1)) < 0.00000001) {
                --result;
            }
        }
        System.out.println(result);
    }
}

File: project_euler/Problem029/Problem029.py
import math

class Problem029:
    def solution(self) -> None:
        start, end, terms = 2, 100, []
        for i in range(start, end + 1):
            for j in range(start, end + 1):
                terms.append(i * math.log(j))

        terms.sort()

        result = len(terms)
        for i in range(result - 1, 0, -1):
            if abs(terms[i] - terms[i - 1]) < 0.00000001:
                result -= 1
        print(result)
